datastorages/jcloud/src/main/java/org/fao/geonet/api/records/attachments/JCloudStore.java

```diff
@@ -254,7 +347,8 @@ public String delResources(final ServiceContext context, final int metadataId) t
                 PageSet<? extends StorageMetadata> page = jCloudConfiguration.getClient().getBlobStore().list(jCloudConfiguration.getContainerName(), opts);
 
                 for (StorageMetadata storageMetadata : page) {
-                    if (storageMetadata.getType() == StorageType.BLOB) {
+                    // For azure Blob if the type is folder then tye storage type will be BLOB and hdi_isfolder=true so we cannot only rely on StorageType.FOLDER
+                    if (storageMetadata.getType().equals(StorageType.BLOB) && !"true".equals(storageMetadata.getUserMetadata().get("hdi_isfolder")))  {
```

Comment:
I would move the string `hdi_isfolder` to a constant, with some documentation.

Reply:
Ok,, but I don't see any change yet, I guess you are working on the change?

Reply:
Odd that you don't see the changes

This is the new static field
https://github.com/geonetwork/core-geonetwork/blob/d7ec517527f3e44f069bbadd0c249af76cf19ab8/datastorages/jcloud/src/main/java/org/fao/geonet/api/records/attachments/JCloudStore.java#L79

This is the new code.
https://github.com/geonetwork/core-geonetwork/blob/d7ec517527f3e44f069bbadd0c249af76cf19ab8/datastorages/jcloud/src/main/java/org/fao/geonet/api/records/attachments/JCloudStore.java#L352-L356

And this is the new `isFolder()` function
https://github.com/geonetwork/core-geonetwork/blob/d7ec517527f3e44f069bbadd0c249af76cf19ab8/datastorages/jcloud/src/main/java/org/fao/geonet/api/records/attachments/JCloudStore.java#L503-L506

Reply:
Maybe I had an old version in the browser, I see it has been done a push force, but now I see the changes, thanks.